### PR TITLE
[1LP][RFR] Require uncollect reason

### PR DESCRIPTION
**Pull request by mshriver**
Make reason required for uncollect/uncollectif markers, raising a ValueError during collection if its not included.

Also update all of the current uses that don't have a reason.

During collection, this results in an INTERNALERROR that will look like the following at the bottom of the traceback:

```
INTERNALERROR>   File "/home/mshriver/repos/integration_tests/cfme/markers/uncollect.py", line 141, in pytest_collection_modifyitems
INTERNALERROR>     raise ValueError(REASON_REQUIRED.format(item.name))
INTERNALERROR> ValueError: A reason is required when using uncollect(if): test_upgrade_single_inplace

```

### CODE REVIEW

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 19f2e34da25faced5133637e396b3626ee139948

The *local* fixture **`button_group`** is used in the following files:
- **cfme/tests/automate/custom_button/test_buttons.py**
    - *test_button_crud*
    - *test_attribute_override*
    - *test_under_group_multiple_button_crud*

The *local* fixture **`setup_obj`** is used in the following files:
- **cfme/tests/automate/custom_button/test_infra_objects_ansible.py**

The *local* fixture **`user_self_service_role`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**

The *local* fixture **`vis_enb_button`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**

The *local* fixture **`host`** is used in the following files:
- **cfme/tests/candu/test_cluster_graph.py**
    - *test_cluster_graph_screen*

The *local* fixture **`start_evmserverd_after_module`** is used in the following files:
- **cfme/tests/cli/test_evmserverd.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit f4ef536004a51d0fd41d9c314f1ca2d9e93981f2

The *local* fixture **`button_group`** is used in the following files:
- **cfme/tests/automate/custom_button/test_buttons.py**
    - *test_button_crud*
    - *test_attribute_override*
    - *test_under_group_multiple_button_crud*

The *local* fixture **`setup_obj`** is used in the following files:
- **cfme/tests/automate/custom_button/test_infra_objects_ansible.py**

The *local* fixture **`user_self_service_role`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**

The *local* fixture **`vis_enb_button`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**

The *local* fixture **`host`** is used in the following files:
- **cfme/tests/candu/test_cluster_graph.py**
    - *test_cluster_graph_screen*

The *local* fixture **`start_evmserverd_after_module`** is used in the following files:
- **cfme/tests/cli/test_evmserverd.py**

The *local* fixture **`vm_crud`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_genealogy.py**

The *local* fixture **`set_landing_page`** is used in the following files:
- **cfme/tests/configure/test_landing_page.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 40db2e234d1f67045c962cde5d0e5cecccd84491

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *local* fixture **`button_group`** is used in the following files:
- **cfme/tests/automate/custom_button/test_buttons.py**
    - *test_button_crud*
    - *test_attribute_override*
    - *test_under_group_multiple_button_crud*

The *local* fixture **`setup_obj`** is used in the following files:
- **cfme/tests/automate/custom_button/test_infra_objects_ansible.py**

The *local* fixture **`user_self_service_role`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**

The *local* fixture **`vis_enb_button`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**

The *local* fixture **`host`** is used in the following files:
- **cfme/tests/candu/test_cluster_graph.py**
    - *test_cluster_graph_screen*

The *local* fixture **`start_evmserverd_after_module`** is used in the following files:
- **cfme/tests/cli/test_evmserverd.py**

The *local* fixture **`vm_crud`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_genealogy.py**

The *local* fixture **`set_landing_page`** is used in the following files:
- **cfme/tests/configure/test_landing_page.py**

The *local* fixture **`host_with_credentials`** is used in the following files:
- **cfme/tests/infrastructure/test_host_analysis.py**

The *local* fixture **`log_monitor`** is used in the following files:
- **cfme/tests/integration/test_cfme_auth.py**

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 1984bb8b533b14bee29711786a5f10d067687efb

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *local* fixture **`button_group`** is used in the following files:
- **cfme/tests/automate/custom_button/test_buttons.py**
    - *test_button_crud*
    - *test_attribute_override*
    - *test_under_group_multiple_button_crud*

The *local* fixture **`setup_obj`** is used in the following files:
- **cfme/tests/automate/custom_button/test_infra_objects_ansible.py**

The *local* fixture **`user_self_service_role`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**

The *local* fixture **`vis_enb_button`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**

The *local* fixture **`host`** is used in the following files:
- **cfme/tests/candu/test_cluster_graph.py**
    - *test_cluster_graph_screen*

The *local* fixture **`start_evmserverd_after_module`** is used in the following files:
- **cfme/tests/cli/test_evmserverd.py**

The *local* fixture **`vm_crud`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_genealogy.py**

The *local* fixture **`set_landing_page`** is used in the following files:
- **cfme/tests/configure/test_landing_page.py**

The *local* fixture **`host_with_credentials`** is used in the following files:
- **cfme/tests/infrastructure/test_host_analysis.py**

The *local* fixture **`log_monitor`** is used in the following files:
- **cfme/tests/integration/test_cfme_auth.py**

The *local* fixture **`local_user`** is used in the following files:
- **cfme/tests/integration/test_cfme_auth.py**
    - *test_login_local_group*

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

File: cfme/markers/uncollect.py [hunk 1]
```diff
@@ -77,15 +80,16 @@ def uncollectif(item):
 
     from cfme.utils.pytest_shortcuts import extract_fixtures_values
     for _, mark in item.iter_markers_with_node('uncollectif') or []:
-        log_msg = 'Trying uncollecting {}: {}'.format(
-            item.name,
-            mark.kwargs.get('reason', 'No reason given'))
+        reason = mark.kwargs.get('reason')
+        if reason is None:
+            raise ValueError(REASON_REQUIRED.format(item.name))
```
Comment (digitronik) [inline cfme/markers/uncollect.py]: :+1: Thanks for it

**Comment by mshriver** [comment]
@izapolsk thanks! Not too worried about PRT failures, if they aren't coming from bad test collection ;)

**Comment by jawatts** [comment]
:+1:

**Pull request merged**